sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuWindowExec.scala

```diff
@@ -1483,35 +1458,57 @@ class GpuRunningWindowIterator(
     }
   }
 
-  def computeRunning(cb: ColumnarBatch): ColumnarBatch = {
+  def computeRunning(input: ColumnarBatch): ColumnarBatch = {
     val fixers = fixerIndexMap
-    val numRows = cb.numRows()
-
-    withResource(computeBasicWindow(cb)) { basic =>
-      withResource(GpuProjectExec.project(cb, boundPartitionSpec)) { parts =>
-        val partColumns = GpuColumnVector.extractBases(parts)
-        withResourceIfAllowed(arePartsEqual(lastParts, partColumns)) { partsEqual =>
-          val fixedUp = if (fixerNeedsOrderMask) {
-            withResource(GpuProjectExec.project(cb, boundOrderColumns)) { order =>
-              val orderColumns = GpuColumnVector.extractBases(order)
-              // We need to fix up the rows that are part of the same batch as the end of the
-              // last batch
-              withResourceIfAllowed(areOrdersEqual(lastOrder, orderColumns, partsEqual)) {
-                orderEqual =>
-                  closeOnExcept(fixUpAll(basic, fixers, partsEqual, Some(orderEqual))) { fixed =>
-                    saveLastOrder(getScalarRow(numRows - 1, orderColumns))
-                    fixed
+    val numRows = input.numRows()
+
+    val cbSpillable = SpillableColumnarBatch(input, SpillPriorities.ACTIVE_BATCHING_PRIORITY)
+    withRetryNoSplit(cbSpillable) { _ =>
+      // note that computeBasicWindowSpillable also creates a spillable
+      // batch based on a projection applied to cbSpillable so this probably
+      // needs more work to avoid creating two spillable batches?
+      withResource(computeBasicWindow(cbSpillable.getColumnarBatch())) { basic =>
+        // we backup the fixers state and restore it in the event of a retry
+        var newOrder: Option[Array[Scalar]] = None
+        var newParts: Option[Array[Scalar]] = None
+        val fixedUp = withRestoreOnRetry(fixers.values.toSeq) {
+          withResource(GpuProjectExec.project(cbSpillable.getColumnarBatch(),
+            boundPartitionSpec)) { parts =>
+            val partColumns = GpuColumnVector.extractBases(parts)
+
+            withResourceIfAllowed(arePartsEqual(lastParts, partColumns)) { partsEqual =>
+              val fixedUp = if (fixerNeedsOrderMask) {
+                withResource(GpuProjectExec.project(cbSpillable.getColumnarBatch(),
+                  boundOrderColumns)) { order =>
+                  val orderColumns = GpuColumnVector.extractBases(order)
+                  // We need to fix up the rows that are part of the same batch as the end of the
+                  // last batch
+                  withResourceIfAllowed(areOrdersEqual(lastOrder, orderColumns, partsEqual)) {
+                    orderEqual =>
+                      closeOnExcept(fixUpAll(basic, fixers, partsEqual, Some(orderEqual))) {
+                        fixedUp =>
+                          newOrder = Some(getScalarRow(numRows - 1, orderColumns))
```

Comment:
Do we care about this leaking if a retry is thrown after this? I think it would only happen if newParts gets an OOM it cannot recover from. It is probably okay, but would be good to at least put in a comment here and newParts describing why it is safe so if someone refactors the code it will still be okay.

Reply:
I've now added a try/catch so that we release any interim results in the case of an exception